x/mint/abci.go

```diff
@@ -11,11 +13,22 @@ func BeginBlocker(ctx sdk.Context, k Keeper) {
 	minter := k.GetMinter(ctx)
 	params := k.GetParams(ctx)
 
+	// Given we cannot calculate average blocktime we dont mint on block 1.
+	if ctx.BlockHeight() == 1 || minter.LastBlockTimestamp.IsZero() {
+		minter.LastBlockTimestamp = ctx.BlockHeader().Time
+		k.SetMinter(ctx, minter)
+
+		return
+	}
+
+	minter.AverageBlockTime = getNewAverageBlockTime(ctx, params, minter)
```

Comment:
Since we pass in `minter` and also set a value on it, it makes sense to instead have a method on `Minter`:

```go
func (m Minter) getNewAverageBlockTime(ctx, params) {}
```